src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -73,6 +73,36 @@ public static void ResetColor()
             }
         }
 
+        public static int WindowWidth
+        {
+            get
+            {
+                if (!ConsoleOutIsTerminal) return 0; // Should we rather throw exception here? On windows we get IOException from the native code, do we need to match for code portability?
+                return TerminalBasicInfo.Instance.ColumnFormat;
```

Comment:
I checked tput cols changes on resizing and was under the impression that TermInfo db and tput give the same result but seems like I was wrong. Hmm.. the only other way I know this can be done is using Ioctl and tput cols and none is POSIX compliant.